src/logging.cpp

```diff
@@ -29,45 +34,169 @@ bl::sources::severity_logger<int> warning(3);  // Strange events
 bl::sources::severity_logger<int> error(4);  // Recoverable errors
 bl::sources::severity_logger<int> fatal(5);  // Unrecoverable errors
 
-/**
- * @brief Flush the log.
- *
- * EXAMPLES:
- * ```cpp
- * log_flush();
- * ```
- */
-void
-log_flush() {
-  sink->flush();
-}
+struct NoDelete {
+  void
+  operator()(void *) {}
+};
 
-/**
- * @brief Print help to stdout.
- * @param name The name of the program.
- *
- * EXAMPLES:
- * ```cpp
- * print_help("sunshine");
- * ```
- */
-void
-print_help(const char *name) {
-  std::cout
-    << "Usage: "sv << name << " [options] [/path/to/configuration_file] [--cmd]"sv << std::endl
-    << "    Any configurable option can be overwritten with: \"name=value\""sv << std::endl
-    << std::endl
-    << "    Note: The configuration will be created if it doesn't exist."sv << std::endl
-    << std::endl
-    << "    --help                    | print help"sv << std::endl
-    << "    --creds username password | set user credentials for the Web manager"sv << std::endl
-    << "    --version                 | print the version of sunshine"sv << std::endl
-    << std::endl
-    << "    flags"sv << std::endl
-    << "        -0 | Read PIN from stdin"sv << std::endl
-    << "        -1 | Do not load previously saved state and do retain any state after shutdown"sv << std::endl
-    << "           | Effectively starting as if for the first time without overwriting any pairings with your devices"sv << std::endl
-    << "        -2 | Force replacement of headers in video stream"sv << std::endl
-    << "        -p | Enable/Disable UPnP"sv << std::endl
-    << std::endl;
-}
+BOOST_LOG_ATTRIBUTE_KEYWORD(severity, "Severity", int)
+
+namespace logging {
+  /**
+   * @brief A destructor that restores the initial state.
+   */
+  deinit_t::~deinit_t() {
+    deinit();
+  }
+
+  /**
+   * @brief Deinitialize the logging system.
+   *
+   * EXAMPLES:
+   * ```cpp
+   * deinit();
+   * ```
+   */
+  void
+  deinit() {
+    log_flush();
+    bl::core::get()->remove_sink(sink);
+    sink.reset();
+  }
+
+  /**
+   * @brief Initialize the logging system.
+   * @param min_log_level The minimum log level to output.
+   * @param log_file The log file to write to.
+   * @returns A deinit_t object that will deinitialize the logging system when it goes out of scope.
+   *
+   * EXAMPLES:
+   * ```cpp
+   * log_init(2, "sunshine.log");
+   * ```
+   */
+  [[nodiscard]] std::unique_ptr<deinit_t>
+  init(int min_log_level, const std::string &log_file) {
```

Comment:
Maybe it would be a good idea to either do nothing if `sink != nullptr` or call `deinit` to reinitialize the whole logger?